is_core/forms/widgets.py

```diff
@@ -375,3 +375,21 @@ def render(self, name, value, attrs=None):
     def value_from_datadict(self, data, files, name):
         value = super(MultipleTextInput, self).value_from_datadict(data, files, name)
         return [v.strip() for v in value.split(self.separator)] if isinstance(value, six.string_types) else value
+
+
+class AbstractDateRangeWidget(MultiWidget):
+
+    def __init__(self, filter_name):
+        class_from, class_to = self.get_range_classes()
+        super(AbstractDateRangeWidget, self).__init__(
+            (
+                TextInput(attrs={'class': class_from, 'data-filter': '{}__gte'.format(filter_name)}),
```

Comment:
pokud to dam predbezne nejak takhle:

class DateRangeFromWidget(TextInput):
    
def render(self, name, value, attrs=None):
    '''
    name = napr 'filter__created_at_0'; 
    '''   
    if value is None:
            value = ''
        final_attrs = self.build_attrs(type=self.input_type, name=name)
        if value != '':
            # Only add the 'value' attribute if a value is non-empty.
            final_attrs['value'] = force_text(self.format_value(value))
        return format_html('<input{} />', flatatt(final_attrs))

    def build_attrs(self, **kwargs):
        attrs = super().build_attrs(**kwargs)
        attrs['data-filter'] = '{}__{}'.format(**attrs['data-filter']**, 'gte')
        return attrs

class DateRangeToWidget(TextInput):

    def render(self, name, value, attrs=None):
        if value is None:
            value = ''
        final_attrs = self.build_attrs(type=self.input_type, name=name)
        if value != '':
            # Only add the 'value' attribute if a value is non-empty.
            final_attrs['value'] = force_text(self.format_value(value))
        return format_html('<input{} />', flatatt(final_attrs))

    def build_attrs(self, **kwargs):
        attrs = super().build_attrs(**kwargs)
        attrs['data-filter'] = '{}__{}'.format(attrs['data-filter'], 'lte')
        return attrs


class AbstractDateRangeWidget(MultiWidget):

    def __init__(self):
        class_from, class_to = self.get_range_classes()
        super(AbstractDateRangeWidget, self).__init__(
            (
                DateRangeFromWidget(),
                DateRangeToWidget(),
            )
        )

tak 'attrs['data-filter']' zarve error
attrs['data-filter'] = '{}__{}'.format(attrs['data-filter'], 'gte')
KeyError: 'data-filter'